---
title: "Lab16_JohnBauer"
author: "John Bauer"
date: "2024-05-1"
output: 
  html_document:
    toc: true
    toc_depth: 4
    Theme: yeti
    Highlight: haddock
    code_folding: hide
---
## Library
```{r}
library(tidyverse)
library(knitr)
library(ggtree)
library(TDbook) #A Companion Package for the Book "Data Integration, Manipulation and Visualization of Phylogenetic Trees" by Guangchuang Yu (2022, ISBN:9781032233574).
library(ggimage)
library(rphylopic)
library(treeio)
library(tidytree)
library(ape)
library(TreeTools)
library(phytools)
library(ggnewscale)
library(ggtreeExtra)
library(ggstar)
library(ggeasy) #https://jonocarroll.github.io/ggeasy/
library(DT)
```
## NEON_MAGs
```{r, message=FALSE, warning=FALSE}
NEON_MAGs <- read_csv("data/NEON/GOLD_Study_ID_Gs0161344_NEON_2024_4_21.csv") %>% 
  # remove columns that are not needed for data analysis
  select(-c(`GOLD Study ID`, `Bin Methods`, `Created By`, `Date Added`, `Bin Lineage`)) %>% 
  # create a new column with the Assembly Type
  mutate("Assembly Type" = case_when(`Genome Name` == "NEON combined assembly" ~ `Genome Name`,
                            TRUE ~ "Individual")) %>% 
  mutate_at("Assembly Type", str_replace, "NEON combined assembly", "Combined") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "d__", "") %>%  
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "p__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "c__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "o__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "f__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "g__", "") %>% 
  mutate_at("GTDB-Tk Taxonomy Lineage", str_replace, "s__", "") %>%
  separate(`GTDB-Tk Taxonomy Lineage`, c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), ";", remove = FALSE) %>% 
  mutate_at("Domain", na_if,"") %>% 
  mutate_at("Phylum", na_if,"") %>% 
  mutate_at("Class", na_if,"") %>% 
  mutate_at("Order", na_if,"") %>% 
  mutate_at("Family", na_if,"") %>% 
  mutate_at("Genus", na_if,"") %>% 
  mutate_at("Species", na_if,"") %>% 
  
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "S-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-") 
```

## NEON_metagenomes + NEON_chemistry for gamma and toolik, Tree_bac 
```{r, message=FALSE, warning=FALSE}
NEON_metagenomes <- read_tsv("data/NEON/exported_img_data_Gs0161344_NEON.tsv") %>% 
  select(-c(`Domain`, `Sequencing Status`, `Sequencing Center`)) %>% 
  rename(`Genome Name` = `Genome Name / Sample Name`) %>% 
  filter(str_detect(`Genome Name`, 're-annotation', negate = T)) %>% 
  filter(str_detect(`Genome Name`, 'WREF plot', negate = T)) 

NEON_metagenomes <- NEON_metagenomes %>% 
  # Get rid of the the common string "Soil microbial communities from "
  mutate_at("Genome Name", str_replace, "Terrestrial soil microbial communities from ", "") %>% 
  # Use the first `-` to split the column in two
  separate(`Genome Name`, c("Site","Sample Name"), " - ") %>% 
  # Get rid of the the common string "-comp-1"
  mutate_at("Sample Name", str_replace, "-comp-1", "") %>%
  # separate the Sample Name into Site ID and plot info
  separate(`Sample Name`, c("Site ID","subplot.layer.date"), "_", remove = FALSE,) %>% 
  # separate the plot info into 3 columns
  separate(`subplot.layer.date`, c("Subplot", "Layer", "Date"), "-")

NEON_chemistry <- read_tsv("data/NEON/neon_plot_soilChem1_metadata.tsv") %>% 
  # remove -COMP from genomicsSampleID
  mutate_at("genomicsSampleID", str_replace, "-COMP", "") 

NEON_MAGs_metagenomes_chemistry <- NEON_MAGs %>% 
  left_join(NEON_metagenomes, by = "Sample Name") %>% 
  left_join(NEON_chemistry, by = c("Sample Name" = "genomicsSampleID")) %>%
  rename("label" = "Bin ID")

tree_arc <- read.tree("data/NEON/gtdbtk.ar53.decorated.tree")
tree_bac <- read.tree("data/NEON/gtdbtk.bac120.decorated.tree")

#Gammaproteobacteria
NEON_MAGs_metagenomes_chemistry_Gammaproteobacteria <- NEON_MAGs_metagenomes_chemistry %>%
  filter(str_detect(`Class`,"Gammaproteobacteria"))

#Novel
NEON_MAGs_metagenomes_chemistry_Gammaproteobacteria_Novel <- NEON_MAGs_metagenomes_chemistry_Gammaproteobacteria %>%
  filter(is.na(Order) | is.na(Family) | is.na(Genus) | is.na(Species))
#almost are novel only two have species names


#Toolik
NEON_MAGs_metagenomes_chemistry_TOOL<- NEON_MAGs_metagenomes_chemistry %>% 
  filter(`Site ID.x` == "TOOL")
NEON_MAGs_metagenomes_chemistry_TOOL_Novel <- NEON_MAGs_metagenomes_chemistry_TOOL %>%
  filter(is.na(Order) | is.na(Family) | is.na(Genus) | is.na(Species))
#almost are novel only two have species names
TOOL_MAGs_label <- NEON_MAGs_metagenomes_chemistry_TOOL$label
tree_bac_TOOL_MAGs <-drop.tip(tree_bac,tree_bac$tip.label[-match(TOOL_MAGs_label, tree_bac$tip.label)])

# Make a vector with the internal node lables
node_vector_bac_TOOL_MAGS = c(tree_bac_TOOL_MAGs$tip.label,tree_bac_TOOL_MAGs$node.label)

```

## Finding notables
```{r}
#finding notable gammaproteobacteria
datatable(NEON_MAGs_metagenomes_chemistry_TOOL %>%
          count(Order, sort=TRUE))

#steroidobacterales, Burkholderiales, xanthomodales
datatable(NEON_MAGs_metagenomes_chemistry_TOOL %>%
          count(Genus, sort=TRUE))
#caballeronia, rhodanobacter, herbspirillum, paraburkholderia
datatable(NEON_MAGs_metagenomes_chemistry_TOOL %>%
          count(Species, sort=TRUE))
#Novel bacteria in the Class Gammaproteobacteria
datatable (NEON_MAGs_metagenomes_chemistry_TOOL_Novel %>%
             count(Order, sort=TRUE))

```

## Find node labels
```{r}
# Make a vector with the internal node labels
node_vector_bac = c(tree_bac$tip.label,tree_bac$node.label)

# Search for your Phylum or Class
grep("Gammaproteobacteria", node_vector_bac, value = TRUE)
match(grep("Gammaproteobacteria", node_vector_bac, value = TRUE), node_vector_bac)

grep("Alphaproteobacteria", node_vector_bac, value = TRUE)
match(grep("Alphaproteobacteria", node_vector_bac, value = TRUE), node_vector_bac)

grep("Pseudomonadota", node_vector_bac, value = TRUE)
match(grep("Pseudomonadota", node_vector_bac, value = TRUE), node_vector_bac)

tree_bac_node_Gammaproteobacteria <- Preorder(tree_bac)
tree_Gammaproteobacteria <- Subtree(tree_bac_node_Gammaproteobacteria, 3048)
```


# Graphs

## Toolik Bin Contamination  vs Bin Completeness
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry_TOOL %>%
     ggplot(aes(y=`Bin Contamination`,x=`Bin Completeness`, color=`Bin Quality`))+geom_point()+labs(title="Toolik Sample Bin Completeness vs Bin Contamination", y="Bin Contamination", x="Bin Completeness")+theme_classic()
ggsave(
  "Toolik_Sample_Soil_Bin_Completeness_vs_Bin_Contamination.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Bin Contamination  vs Bin Completeness compared to Total population
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry %>%
     ggplot(aes(y=`Bin Contamination`,x=`Bin Completeness`, color=`Bin Quality`))+geom_point(shape=20)+geom_point(data=NEON_MAGs_metagenomes_chemistry_TOOL, aes(y=`Bin Contamination`,x=`Bin Completeness`), shape=18, size=5)+labs(title="Toolik Sample Bin Completeness vs Bin Contamination", y="Bin Contamination", x="Bin Completeness")+theme_classic()
ggsave(
  "Toolik_Sample_Soil_Bin_Completeness_vs_Bin_Contamination_Total_Population_Comparison.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Sample Soil Temperature vs pH in Water
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry_TOOL %>%
     ggplot(aes(y=`soilTemp`,x=`soilInWaterpH`, color=Genus))+geom_point()+labs(title="Toolik Sample Soil Temperature vs pH in Water", y="Soil Temperature", x="Soil pH in Water")+theme_classic()
ggsave(
  "Toolik_Sample_Soil_Temperature_vs_pH_in_Water.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Sample Elevation vs Family 
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry_TOOL %>%
     ggplot(aes(y=as.integer(`Elevation In Meters`),x=`Family`, color=Order))+geom_point()+labs(title="Toolik Sample Elevation vs Family", y="Elevation (m)", x="Family")+theme_classic()+theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1))
ggsave(
  "Toolik_Sample_Elevation_vs_Family.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Sample Longitude vs Family
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry_TOOL %>%
     ggplot(aes(y=`Longitude`,x=`Family`, color=Order))+geom_point()+labs(title="Toolik Sample Longitude vs Family", y="Longitude", x="Family",x.offset=2)+theme_classic()+theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1))
ggsave(
  "Toolik_Sample_Longitude_vs_Family.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Sample Lattitude vs Family
```{r, fig.width=12, fig.height=12}
NEON_MAGs_metagenomes_chemistry_TOOL %>%
     ggplot(aes(y=`Latitude`,x=`Family`, color=Order))+geom_point()+labs(title="Toolik Sample Latitude vs Family", y="Lattitude", x="Family",x.offset=2)+theme_classic()+theme(axis.text.x=element_text(angle=45, vjust=1, hjust=1))
ggsave(
  "Toolik_Sample_Lattitude_vs_Family.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

# Trees

## Toolik Rectangular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs) +
  geom_tiplab(size=3) + xlim(0,5)
  
ggsave(
  "Toolik_Rectangular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)


```

## Toolik Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular") + 
  geom_tiplab(aes(angle=angle))+
    theme_tree() +
    xlim(0,1.5)

ggsave(
  "Toolik_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

```

## Toolik Circular Tree with phylum color
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry +
  geom_point(mapping=aes(color=Phylum))
ggsave(
  "NEON_MAGS_Toolik_Circular_Tree_with_Phylum_color.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Gammaproteobacteria Highlight Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular") +
    geom_hilight(node=259, fill="steelblue", alpha=.6)

ggsave(
  "Toolik_Gammaproteobacteria_Highlight_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Gammaproteobacteria Highlight with Label Circular Tree
```{r, fig.height=14, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular", branch.length="none") +
    geom_hilight(node=258, fill="grey", alpha=.6) +
    geom_cladelab(node=258, label="Pseudomonadota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='grey',barsize=1.5, fontsize=5)+
    geom_hilight(node=259, fill="steelblue", alpha=.6) +
    geom_cladelab(node=259, label="Gammaproteobacteria", align=TRUE, angle='auto', offset=0.75,  
                  offset.text=0.5 , textcolor='black', barcolor='steelblue',barsize=1.5, fontsize=5)

ggsave(
  "Toolik_Gammaproteobacteria_Highlight_with_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Find node
```{r}
datatable(NEON_MAGs_metagenomes_chemistry_TOOL %>%
            count(Phylum, sort=TRUE))
grep("Phycisphaerae", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Phycisphaerae", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Acidobacteriota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Acidobacteriota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Actinomycetota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Actinomycetota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Myxococcota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Myxococcota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Bacteroidota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Bacteroidota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Verrucomicrobiota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Verrucomicrobiota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Chloroflexota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Chloroflexota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)



grep("Eremiobacterota", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Eremiobacterota", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Patescibacteria", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Patescibacteria", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

grep("Patescibacteria", node_vector_bac_TOOL_MAGS, value = TRUE)
match(grep("Patescibacteria", node_vector_bac_TOOL_MAGS, value = TRUE), node_vector_bac_TOOL_MAGS)

```

## Toolik with highlights for each phylum
```{r, fig.height=14, fig.width=12}
NEON_MAGs_metagenomes_chemistry_TOOL_Named <-NEON_MAGs_metagenomes_chemistry_TOOL %>%
  filter(!is.na(Species))

ggtree(tree_bac_TOOL_MAGs, layout="circular", branch.length="none")+
    geom_hilight(node=219, fill="purple", alpha=.6) +
    geom_cladelab(node=219, label="Acidobacteriota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='purple',barsize=1.5, fontsize=5)+
    geom_hilight(node=258, fill="grey", alpha=.6) +
    geom_cladelab(node=258, label="Pseudomonadota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='grey',barsize=1.5, fontsize=5)+
    geom_hilight(node=157, fill="green", alpha=.6) +
    geom_cladelab(node=157, label="Actinomycetota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='green',barsize=1.5, fontsize=5)+
    geom_hilight(node=247, fill="red", alpha=.6) +
    geom_cladelab(node=247, label="Myxococcota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='red',barsize=1.5, fontsize=5)+
    geom_hilight(node=204, fill="yellow", alpha=.6) +
    geom_cladelab(node=204, label="Bacteroidota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='yellow',barsize=1.5, fontsize=5)+
    geom_hilight(node=151, fill="orange", alpha=.6) +
    geom_cladelab(node=151, label="Chloroflexota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='orange',barsize=1.5, fontsize=5)+
    geom_hilight(node=211, fill="brown", alpha=.6) +
    geom_cladelab(node=211, label="Planctomycetota", align=TRUE, angle='auto', 
                  offset.text=0.5 , textcolor='black', barcolor='brown',barsize=1.5, fontsize=5)+
    geom_hilight(node=213, fill="blue", alpha=.6) +
    geom_cladelab(node=213, label="Verrucomicrobiota", align=TRUE, angle='auto', offset=0.75,  
                  offset.text=0.5 , textcolor='black', barcolor='blue',barsize=1.5, fontsize=5)+ 


ggsave(
  "Toolik_Phylum_Highlight_with_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Gammaproteobacteria Label Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular", branch.length="none") %>% 
  collapse(node=259) + 
  geom_point2(aes(subset=(node==259)), shape=23, size=5, fill='steelblue') +  
  geom_cladelab(node=259, label="Gammaproteobacteria", align=TRUE,  
                  offset = 2, textcolor='steelblue')
ggsave(
  "Toolik_Gammaproteobacteria_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik Gammaproteobacteria Minimized Circular Tree
```{r, fig.height=12, fig.width=12}
p <- ggtree(tree_bac_TOOL_MAGs, layout="circular", branch.length="none")
scaleClade(p, 259, .2) %>% collapse(259, 'min', fill="steelblue")  +
  geom_cladelab(node=259, label="Gammaproteobacteria", align=TRUE,  
                  offset = 0.2, textcolor='steelblue')
ggsave(
  "Toolik_MAGS_Gammaproteobacteria_Minimized_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik MAGS with Order Labels Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tiplab(size=2, hjust=-.1) +
  geom_point(mapping=aes(color=Order))
ggsave(
  "Toolik_MAGS__with_Order_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik with Order and Assembly Type Labels Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tiplab(size=2, hjust=-.1) +
  geom_point(mapping=aes(color=Order, shape = `Assembly Type`)) 
ggsave(
  "Toolik_MAGS_with_Order_and_Assembly_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik with Ecosystem Subtype System Rectangular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tiplab(size=2, hjust=-.1) +
  geom_point(mapping=aes(color=`Ecosystem Subtype`))
ggsave(
  "Toolik_MAGS_with_Ecosystem_Subtype_System_Rectangular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik with Bin Completeness Label Rectangular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tippoint(aes(colour=`Bin Completeness`)) + 
  scale_colour_gradient(low='blue', high='red') +
  geom_tiplab(size=2, hjust=-.1) 
  
ggsave(
  "Toolik_MAGS_with_Bin_Completeness_Label_Rectangular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik with Ecosystem Subtype and Total number of Bases Label Circular Tree
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular")  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`))
ggsave(
  "Toolik_MAGS_with_Ecosystem_Subtype_and_Total_Number_of_Bases_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik triplot with Bin Completeness and Bin Contamination
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tippoint(aes(colour=`Ecosystem Subtype`)) + 

# For unknown reasons the following does not like blank spaces in the names
  geom_facet(panel = "Bin Completeness", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_point, 
      mapping=aes(x = BinCompleteness)) +
  geom_facet(panel = "Bin Contamination", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_col, 
                aes(x = BinContamination), orientation = 'y', width = .6) +
  theme_tree2(legend.position=c(.1, .7))

ggsave(
  "Toolik_MAGS_triplot_with_Bin_Completeness_and_Bin_Contamination.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik triplot with Gene and GC Content
```{r, fig.height=12, fig.width=12}
ggtree(tree_bac_TOOL_MAGs)  %<+%
  NEON_MAGs_metagenomes_chemistry + 
  geom_tippoint(aes(color=`Ecosystem Subtype`)) + 

# For unknown reasons the following does not like blank spaces in the names
  geom_facet(panel = "Gene Count", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_point, 
      mapping=aes(x = GeneCount))+
  geom_facet(panel = "% GC Content ", data = NEON_MAGs_metagenomes_chemistry_noblank, geom = geom_col, 
                aes(x = GCassembled), orientation = 'y', width = .6) +
  theme_tree2(legend.position=c(.1, .7))

ggsave(
  "Toolik_MAGS_triplot_with_ Gene_and_GC_Content.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)
```

## Toolik with Ecosystem Subtype and Total Number of Bases in Outside Labels
```{r,fig.height=12,fig.width=12}
ggtree(tree_bac_TOOL_MAGs, layout="circular", branch.length="none") %<+% 
  NEON_MAGs_metagenomes_chemistry + 
  geom_point2(mapping=aes(color=`Ecosystem Subtype`, size=`Total Number of Bases`)) + 
  new_scale_fill() + 
  geom_fruit(
      data=NEON_MAGs_metagenomes_chemistry_noblank,
      geom=geom_tile,
      mapping=aes(y=label, x=1, fill= AssemblyType),
      offset=0.08,   # The distance between external layers, default is 0.03 times of x range of tree.
      pwidth=0.25 # width of the external layer, default is 0.2 times of x range of tree.
      )
ggsave(
  "Toolik_MAGS_with_Ecosystem_Subtype_and_Total_Number_of_Bases_in_Outside_Label_Circular_Tree.png",
  plot = last_plot(),
  device = NULL,
  path = "/cloud/project/images/Plots/Toolik",
  scale = 1,
  width = NA,
  height = NA,
  units = c("in", "cm", "mm", "px"),
  dpi = 300,
  limitsize = TRUE,
  bg = NULL,
  create.dir = FALSE,
)

```